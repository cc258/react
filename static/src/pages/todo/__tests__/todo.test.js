import React from "react";
import { shallow } from "enzyme";

import Todo from "../Todo";

const setup = () => {
  const props = {
    list: ["123"],
    getData: jest.fn()
  };

  const wrapper = shallow(<Todo {...props} />);
  return {
    props,
    wrapper
  };
};

describe("<Todo />", () => {
  it("Todo toMatchSnapshot", () => {
    const { wrapper } = setup();

    expect(wrapper.find("h1").exists());
    expect(wrapper.find("input")).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("click h1 to test getData", () => {
    const { wrapper, props } = setup();
    wrapper.find("h1").simulate("click");
    expect(props.getData).toBeCalled();
  });

  it("change input value", () => {
    const { wrapper } = setup();

    wrapper
      .find("input")
      .simulate("change", { target: { value: "1234567890" } });
    expect(wrapper.state("newTodo")).toEqual("1234567890");
  });
});
