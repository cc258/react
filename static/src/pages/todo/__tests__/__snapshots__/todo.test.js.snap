// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> Todo toMatchSnapshot 1`] = `ShallowWrapper {}`;
